// // const fn = {
// //     add: (num1, num2) => { return num1 + num2 },
// //     makeUserName: (name) => { return fn.name = name },
// //     makeUserAge: (age) => { return fn.age = age },
// //     throwError: () => {
// //         throw new Error("에러 발생")
// //     },

// // };

// const fn = {
//     add: (num1, num2) => num1 + num2,
//     getName: (callback) => {
//         const name = "Mike";
//         setTimeout(() => {
//             callback(name)
//         }, 3000)
//     },
//     getAge: () => {
//         const age = 30;
//         return new Promise((resolve, reject) => {
//             setTimeout(() => {
//                 resolve(age);
//             }, 3000)

//             // reject('error 메시지가 발생합니다.')
//         })
//     }
// }


// module.exports = fn;

